docs/source/profiles.rst

```diff
@@ -0,0 +1,77 @@
+*********
+Profiling
+*********
+
+.. note::
+
+    This section is for developers, and should be disregarded for production
+    runs and 'normal' users in general.
+
+RADICAL-Pilot allows to tweak the pilot process behavior in many details, and
+specifically allows to artificially increase the load on individual components,
+for the purpose of more detailed profiling, and identification of bottlenecks.
+With that background, a pilot description supports an additional attribute
+`_config`, which accepts a dict of the following structure:
+
+.. code-block:: python
+
+        pdesc = rp.ComputePilotDescription()
+        pdesc.resource = "local.localhost"
+        pdesc.runtime  = 5 # minutes
+        pdesc.cores    = 8
+        pdesc.cleanup  = False
+        pdesc._config  = {'number_of_workers' : {'StageinWorker'   :  1,
+                                                 'ExecWorker'      :  2,
+                                                 'StageoutWorker'  :  1,
+                                                 'UpdateWorker'    :  1},
+                          'blowup_factor'     : {'Agent'           :  1,
```

Comment:
We probably should remove documentation for `pdesc._config` - that should not be exposed to the end user at the moment.

Reply:
Done. I muted the whole section from the index.

Reply:
Makes sense, thanks.